docstr_coverage/printers.py

```diff
@@ -147,3 +164,63 @@ def _print_overall_statistics(self, results):
         )
 
         print_line("Total coverage: {:.1f}%  -  Grade: {}".format(count.coverage(), grade))
+
+
+class JsonPrinter(Printer):
```

Comment:
I assume that the JsonPrinter is primarily used in use cases where the output is interpreted by a machine (which thus allows easy deserialization). We've had such feature requests in the past. 

Thus, the printer should provide the maximum amount of available information. The `results.to_legacy()` only provides an aggregated overview; it would be better to serialize all information present in `results` (which, as opposed to `results.to_legacy()`, may also be extended with additional fields in the future).

If that's too much work, we can of course remove the JsonPrinter in this PR and open a dedicated issue/PR for that.